packages/tests/server/httpSubscriptionLink.test.ts

```diff
@@ -58,14 +61,16 @@ const ctx = konn()
             }),
           )
           .subscription(async function* (opts) {
-            for await (const data of on(ee, 'data', {
+            for await (const [thing] of ee.toIterable('data', {
               signal: opts.signal,
             })) {
-              const thing = data[0] as number | Error;
-
               if (thing instanceof Error) {
                 throw thing;
               }
+              if (thing === returnSymbol) {
+                return;
```

Comment:
is it possible to return a value as the final state? Probably doesn't matter if not but generators I believe can do it so might be expected by some use case, for instance sending a "reason" back?

Reply:
Let's add it if/when people ask for it. I didn't add it here because there's no natural place where this goes on the client without a bit of refactoring (I guess it'd make sense to have it in `onCompleted`)